emcc.py

```diff
@@ -602,10 +602,15 @@ def run(args):
     return 0
 
   if '--version' in args:
-    try:
-      revision = run_process(['git', 'show'], stdout=PIPE, stderr=PIPE, cwd=shared.path_from_root()).stdout.splitlines()[0]
-    except Exception:
-      revision = '(unknown revision)'
+    # if the emscripten folder is not a git repo, don't run git show - that can
+    # look up and find the revision in a parent directory that is a git repo
+    if not os.path.exists(shared.path_from_root('.git')):
+      revision = 'unknown revision'
```

Comment:
I'd rather invert the condition and branches.

Reply:
And invert the comment too `If the emscripten directory contains a .git, attempt to use a git show command to extract a version`.